src/main/scala/com/fulcrumgenomics/util/LogDouble.scala

```diff
@@ -0,0 +1,172 @@
+  def toString(probability: LogDouble): String = toInt(probability=probability).toString
+
+  /** Converts a phred score to a probability (log-space). */
+  def fromPhredScore(phredScore: Double): LogDouble = new LogDouble(logValue = LogOf10 * phredScore / -10)
+
+  implicit class DoubleWrapper(phredScore: Double) {
+    def fromPhredScore: LogDouble = PhredScore.fromPhredScore(phredScore=phredScore)
+  }
+  implicit class IntWrapper(phredScore: Int) {
+    def fromPhredScore: LogDouble = PhredScore.fromPhredScore(phredScore = phredScore)
+  }
+  implicit class LogDoubleWrapper(probability: LogDouble) {
+    def toPhredScore: Double       = PhredScore.toPhredScore(probability=probability)
+    def toPhredScoreInt: Int       = PhredScore.toInt(probability=probability)
+    def toPhredScoreByte: Byte     = PhredScore.toByte(probability=probability)
+    def toPhredScoreChar: Char     = PhredScore.toChar(probability=probability)
+    def toPhredScoreString: String = PhredScore.toString(probability=probability)
+  }
+}
+
+object LogDouble {
+
+  /** log(X) == 0. */
+  val Zero = new LogDouble(logValue=Double.NegativeInfinity)
+
+  /** log(X) == 1. */
+  val One = new LogDouble(logValue=0.0)
+
+  /** Converts the double to log-space. */
+  def toLogDouble(value: Double): LogDouble = {
+    if (value < 0) throw new IllegalArgumentException("Cannot use LogDouble to store values less than zero.")
+    new LogDouble(logValue=log(value))
+  }
+
+  /** Gets the mean of the given phred values. */
+  def mean(logDoubles: LogDouble*): LogDouble = logDoubles.foldLeft(Zero)((sum, qual) => sum + qual) / toLogDouble(logDoubles.length)
+
+  /** Converts a Double to log-space. */
+  implicit class DoubleValue(v: Double) {
+    def toLogDouble: LogDouble = LogDouble.toLogDouble(value=v)
+  }
+
+  /** Converts a string to log-space. */
+  implicit class StringValue(doubleValue: String) {
+    def toLogDouble: LogDouble = LogDouble(doubleValue=doubleValue)
+  }
+
+  /** String constructor when a double (not in log-space) is given. */
+  def apply(doubleValue: String): LogDouble = toLogDouble(value=if (doubleValue.contains(".")) doubleValue.toDouble else doubleValue.toInt)
+
+  /** Convenience constructor. */
+  def apply(logValue: Double): LogDouble = new LogDouble(logValue=logValue)
+}
+
+/**
+  * Represents a Double in log-space, to prevent underflow and overflow.
+  * @param logValue the Double in log-space
+  */
+class LogDouble(val logValue: Double) extends AnyVal with Ordered[LogDouble] {
+  /** Returns the value as a Double. */
+  def linearValue: Double = exp(logValue)
+
+  /** Division. */
+  def /(that: LogDouble): LogDouble = {
+    if (that.logValue == LogDouble.Zero.logValue) throw new IllegalStateException("Trying to divide by 0")
+    new LogDouble(logValue=this.logValue - that.logValue)
+  }
+
+  /** Multiplication. */
+  def * (that: LogDouble): LogDouble = new LogDouble(logValue=this.logValue + that.logValue)
```

Comment:
`LogDouble` is really the standard name, I swear it's not just me.  I am open to calling in `PreciseDouble` of just `Probability`, and then just hide the fact it is in log-scale?

 *http://colorfulengineering.org/logmath-notes.pdf
* https://github.com/scalanlp/breeze/blob/master/math/src/main/scala/breeze/math/LogDouble.scala
* https://en.wikipedia.org/wiki/Log_probability